webapp/sources/rudder/rudder-web/src/test/scala/com/normation/rudder/web/services/ComplianceLineTest.scala

```diff
@@ -0,0 +1,285 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package com.normation.rudder.web.services
+
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.MockCompliance
+import com.normation.rudder.MockDirectives
+import com.normation.rudder.MockGitConfigRepo
+import com.normation.rudder.MockTechniques
+import com.normation.rudder.domain.policies.GlobalPolicyMode
+import com.normation.rudder.domain.policies.PolicyMode.Enforce
+import com.normation.rudder.domain.policies.PolicyModeOverrides
+import com.normation.rudder.domain.policies.PolicyTypeName
+import com.normation.rudder.facts.nodes.CoreNodeFact
+import com.normation.rudder.facts.nodes.QueryContext
+import com.normation.rudder.repository.FullActiveTechniqueCategory
+import com.normation.zio.*
+import org.junit.runner.RunWith
+import org.specs2.mutable.*
+import org.specs2.runner.JUnitRunner
+import scala.collection.MapView
+import scala.util.Random
+
+@RunWith(classOf[JUnitRunner])
+class ComplianceLineTest extends Specification {
+
+  val mockDirectives = new MockDirectives(MockTechniques(new MockGitConfigRepo("")))
+  val mockCompliance = new MockCompliance(mockDirectives)
+  implicit val qc: QueryContext                  = QueryContext.testQC
+  val nodes:       MapView[NodeId, CoreNodeFact] = mockCompliance.nodeFactRepo.getAll().runNow
+  val directives:  FullActiveTechniqueCategory   = mockDirectives.directiveRepo.getFullDirectiveLibrary().runNow
+
+  val stableRandom = new Random(42)
+  def freshName(): String = stableRandom.nextLong().toString
+
+  "compliance lines serialisation" >> {
+    val nodeId = NodeId("n1")
+    val report = mockCompliance.simpleExample.simpleStatusReports(nodeId)
+    val lines  = ComplianceData
+      .getNodeByRuleComplianceDetails(
+        nodeId,
+        report,
+        PolicyTypeName.rudderBase,
+        nodes.toMap,
+        directives,
+        mockCompliance.simpleExample.simpleCustomRules,
+        GlobalPolicyMode(Enforce, PolicyModeOverrides.Always),
+        true
+      )
+      .json(freshName _)
+      .toJsCmd
+
+    lines ===
+    """[{"rule": "R1", "compliance": [[0, 0.0]
+      |, [0, 0.0]
+      |, [1, 100.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |, [0, 0.0]
+      |]
+      |, "compliancePercent": 100.0, "id": "r1", "details": [{"directive": "25. Testing blocks", "id": "directive-techniqueWithBlocks", "techniqueName": "technique with blocks", "techniqueVersion": "1.0", "compliance": [[0, 0.0]
```

Comment:
I would prefer a json semantic equality check (`equalsJsonSemantic`) and having a prettified JSON without margin in tests, it is very hard to edit,copy/paste and diff otherwise